### ISSUE TITLE
Development instructions

### ISSUE THREAD

**Issue by jpspringall** [openmediavault/openmediavault#1458]
**Is your feature request related to a problem? Please describe.**
Not really a feature request, but I'm wondering if there any instructions on how to setup and run OMV in a local development environment?

It would appear that vagrant installs the latest "production" version into a virtualbox image
There is also a buildenvadm.sh but that just seems to build the deb files in docker container

**Describe the solution you'd like**
I am wondering if there is a way to run OMV locally from say within vscode?
So that i can manually make changes and then test them locally?

Thanks

James

**Comment by votdev** [comment]
You are right, a development documentation is really missing. But for the moment applies: Use the source, Like.

If you want to modify files and test them, use the Vagrant box. The source tree is shared at `/home/vagrant/openmediavault/`. You need to copy the files into the root file system manually, e.g. `cp -rv usr/* /usr`. Then you can test your changes. The UI code can be tested from your host system by running `npm start` in the `debs/openmediavault/workbench` directory.

**Comment by jpspringall** [comment]
Awesome Thanks @votdev I was more or less on the right track.
I've done far bit of reading up on vagrant, and think I have figured it out.
Out of interest is there a preference for which version of node we should use?
According to the angular documentation we can use 12 or 14

**Comment by votdev** [comment]
I'm running node v16.18.0 on my host system.

**Comment by jpspringall** [comment]
Thanks will go with that
